modules/es6.promise.js

```diff
@@ -107,7 +107,22 @@ function $resolve(value){
 }
 
 // constructor polyfill
-if(!(isFunction(P) && isFunction(P.resolve) && P.resolve(test = new P(function(){})) == test)){
+var workingPromise = isFunction(P) && isFunction(P.resolve) && P.resolve(test = new P(function(){})) == test;
+// Firefox ~33 had broken subclass support, test that.
+if (workingPromise) {
+  var P2 = function(x) {
+    var self = new P(x);
+    Object.setPrototypeOf(self, P2.prototype);
```

Comment:
In modules insted of `Object.setPrototypeOf` should use `require('./$.set-proto').set` and in browsers w/o `Object.setPrototypeOf` or `__proto__` it's missing.